rlse.config.ts fails to load with TypeScript 6 due to deprecated moduleResolution=node10

## Summary

`rlse.ts@0.0.3` fails to load `rlse.config.ts` when the project uses TypeScript 6.x.

The CLI generates a temporary `tsconfig.json` with `compilerOptions.moduleResolution` set to `"node"`. With TypeScript 6, this is interpreted as the deprecated `node10` mode and fails with `TS5107` unless `ignoreDeprecations: "6.0"` is set.

## Reproduction

In a package using TypeScript `6.0.3` and `rlse.ts@0.0.3`:

```ts
// rlse.config.ts
import { defineConfig, presets } from "rlse.ts";

export default defineConfig(
  presets.npmRelease({
    resolvePackage: { name: "@dathomir/core" },
    calculateNextSemver: { level: "patch" },
    runCommand: "pnpm build",
  }),
);
```

Run:

```sh
pnpm exec rlse --dry-run
```

## Actual Result

```text
.temp/tsconfig.json(1,109): error TS5107: Option 'moduleResolution=node10' is deprecated and will stop functioning in TypeScript 7.0. Specify compilerOption '"ignoreDeprecations": "6.0"' to silence this error.
  Visit https://aka.ms/ts6 for migration information.

ERROR  Error compiling TypeScript file: Command failed: tsc --project .../.temp/tsconfig.json
```

Stack trace points to `importTypeScriptConfig` in `rlse.ts@0.0.3`, which writes the temporary tsconfig.

## Expected Result

`rlse.config.ts` should load successfully under TypeScript 6.x.

## Possible Fix

Use a non-deprecated module resolution value in the generated temporary tsconfig, such as:

```json
{
  "compilerOptions": {
    "moduleResolution": "node16"
  }
}
```

Alternatively, add:

```json
{
  "compilerOptions": {
    "ignoreDeprecations": "6.0"
  }
}
```

But switching away from deprecated `node`/`node10` resolution seems more future-proof.

## Environment

- `rlse.ts`: `0.0.3`
- `typescript`: `6.0.3`
- `node`: `v24.15.0`
- package manager: `pnpm 10.17.1`

--- a/docs/package.json
+++ b/docs/package.json
@@ -31,7 +31,7 @@
     "@types/unist": "^3.0.3",
     "globals": "^15.9.0",
     "hast": "^1.0.0",
-    "typescript": "^5.5.3",
+    "typescript": "^6.0.3",
     "unist": "^0.0.1",
     "unist-util-visit": "^5.0.0",
     "unplugin-fonts": "^1.1.1",

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -67,6 +67,6 @@
     "@types/semver": "^7.5.8",
     "rlse.ts": "workspace:*",
     "tsup": "^8.1.0",
-    "typescript": "^5.2.2"
+    "typescript": "^6.0.3"
   }
 }

--- a/packages/core/src/config/loadRlseConfig.ts
+++ b/packages/core/src/config/loadRlseConfig.ts
@@ -3,10 +3,11 @@ import {
   existsSync,
   mkdirSync,
   readFile,
+  readFileSync,
   rmSync,
   writeFileSync,
 } from "node:fs";
-import { dirname, extname, resolve } from "node:path";
+import { dirname, extname, parse, resolve } from "node:path";
 import { cwd } from "node:process";
 import { promisify } from "node:util";
 import consola from "consola";
@@ -71,8 +72,8 @@ const importTypeScriptConfig = async (filePath: string) => {
       compilerOptions: {
         target: "ESNext",
         useDefineForClassFields: true,
-        module: "commonjs",
-        moduleResolution: "node",
+        module: "Node16",
+        moduleResolution: "node16",
         esModuleInterop: true,
         lib: ["ESNext"],
         skipLibCheck: true,
@@ -96,7 +97,7 @@ const importTypeScriptConfig = async (filePath: string) => {
   );
   writeFileSync(
     resolve(tempDir, "package.json"),
-    JSON.stringify({ type: "commonjs" }),
+    JSON.stringify({ type: getPackageType(filePath) }),
   );
 
   try {
@@ -112,5 +113,30 @@ const importTypeScriptConfig = async (filePath: string) => {
     throw error;
   } finally {
     rmSync(tempDir, { recursive: true, force: true });
+  }
+};
+
+const getPackageType = (filePath: string) => {
+  let currentDir = dirname(filePath);
+  const rootDir = parse(currentDir).root;
+
+  while (true) {
+    const packageJsonPath = resolve(currentDir, "package.json");
+
+    if (existsSync(packageJsonPath)) {
+      const packageJson = JSON.parse(
+        readFileSync(packageJsonPath, "utf-8"),
+      ) as {
+        type?: string;
+      };
+
+      return packageJson.type === "module" ? "module" : "commonjs";
+    }
+
+    if (currentDir === rootDir) {
+      return "commonjs";
+    }
+
+    currentDir = dirname(currentDir);
   }
 };

--- a/packages/core/test/config-version.test.mjs
+++ b/packages/core/test/config-version.test.mjs
@@ -1,6 +1,13 @@
 import assert from "node:assert/strict";
 import { execFileSync, execSync } from "node:child_process";
-import { mkdtempSync, readFileSync, rmSync, writeFileSync } from "node:fs";
+import {
+  mkdirSync,
+  mkdtempSync,
+  readFileSync,
+  rmSync,
+  symlinkSync,
+  writeFileSync,
+} from "node:fs";
 import { tmpdir } from "node:os";
 import path from "node:path";
 import { pathToFileURL } from "node:url";
@@ -209,6 +216,94 @@ export default defineConfig(
     );
 
     assert.equal(packageJson.version, "3.0.0");
+  } finally {
+    rmSync(projectDir, { recursive: true, force: true });
+  }
+});
+
+test("loads TypeScript config", () => {
+  const projectDir = createTempProject();
+
+  try {
+    mkdirSync(path.join(projectDir, "node_modules"), { recursive: true });
+    symlinkSync(packageRoot, path.join(projectDir, "node_modules", "rlse.ts"));
+
+    writeFileSync(
+      path.join(projectDir, "rlse.config.ts"),
+      `import { defineConfig, presets } from "rlse.ts";
+
+export default defineConfig(
+  presets.npmRelease({
+    resolvePackage: { name: "rlse-config-version-fixture" },
+    calculateNextSemver: { version: "4.0.0" },
+    publishNpmPackage: false,
+    commit: false,
+    push: false,
+  }),
+);
+`,
+    );
+
+    execFileSync("node", [cliPath], {
+      cwd: projectDir,
+      stdio: "pipe",
+    });
+
+    const packageJson = JSON.parse(
+      readFileSync(path.join(projectDir, "package.json"), "utf8"),
+    );
+
+    assert.equal(packageJson.version, "4.0.0");
+  } finally {
+    rmSync(projectDir, { recursive: true, force: true });
+  }
+});
+
+test("loads TypeScript config in ESM project", () => {
+  const projectDir = createTempProject();
+
+  try {
+    writeFileSync(
+      path.join(projectDir, "package.json"),
+      JSON.stringify(
+        {
+          name: "rlse-config-version-fixture",
+          version: "1.2.3",
+          type: "module",
+        },
+        null,
+        2,
+      ),
+    );
+    mkdirSync(path.join(projectDir, "node_modules"), { recursive: true });
+    symlinkSync(packageRoot, path.join(projectDir, "node_modules", "rlse.ts"));
+
+    writeFileSync(
+      path.join(projectDir, "rlse.config.ts"),
+      `import { defineConfig, presets } from "rlse.ts";
+
+export default defineConfig(
+  presets.npmRelease({
+    resolvePackage: { name: "rlse-config-version-fixture" },
+    calculateNextSemver: { version: "5.0.0" },
+    publishNpmPackage: false,
+    commit: false,
+    push: false,
+  }),
+);
+`,
+    );
+
+    execFileSync("node", [cliPath], {
+      cwd: projectDir,
+      stdio: "pipe",
+    });
+
+    const packageJson = JSON.parse(
+      readFileSync(path.join(projectDir, "package.json"), "utf8"),
+    );
+
+    assert.equal(packageJson.version, "5.0.0");
   } finally {
     rmSync(projectDir, { recursive: true, force: true });
   }

--- a/packages/core/tsconfig.json
+++ b/packages/core/tsconfig.json
@@ -5,6 +5,7 @@
     "module": "ESNext",
     "lib": ["ESNext", "DOM", "DOM.Iterable", "ES2022"],
     "skipLibCheck": true,
+    "ignoreDeprecations": "6.0",
     "noErrorTruncation": true,
 
     /* Bundler mode */

--- a/playgrounds/vanilla-ts/package.json
+++ b/playgrounds/vanilla-ts/package.json
@@ -18,7 +18,7 @@
   "devDependencies": {
     "rlse.ts": "workspace:*",
     "ts-node": "^10.9.2",
-    "typescript": "^5.2.2",
+    "typescript": "^6.0.3",
     "vite": "^5.3.5"
   }
 }

